pkg/logger/operationid_middleware_test.go

```diff
@@ -0,0 +1,98 @@
+package logger
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func getTestCtxWithOpId() context.Context {
+	return context.WithValue(context.Background(), OpIDKey, testOpId)
+}
+
+func Test_WithOpID(t *testing.T) {
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name     string
+		args     args
+		opIdSet  bool
+		wantOpId string
+	}{
+		{
+			name: "should return context with OpID previously set in the context",
+			args: args{
+				ctx: context.Background(),
+			},
+			opIdSet: false,
+		},
+		{
+			name: "should return context with new OpID if now set in the context",
+			args: args{
+				ctx: getTestCtxWithOpId(),
+			},
+			opIdSet:  true,
+			wantOpId: testOpId,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			context := WithOpID(tt.args.ctx)
+			Expect(context.Value(OpIDKey)).ToNot(Equal(""))
+			Expect(context.Value(OpIDKey) == tt.wantOpId).To(Equal(tt.opIdSet))
+		})
+	}
+}
+
+func Test_GetOperationID(t *testing.T) {
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "should return OpID value when its set in the context",
+			args: args{
+				ctx: context.Background(),
+			},
+			want: "",
+		},
+		{
+			name: "should return an empty string if OpID value is not set in the context",
```

Comment:
Should name of this case be swapped with name of test case in L64 above?

Reply:
yes. I have swapped them. good catch